### ISSUE TITLE
bc-java 1.75 fails to parse (admittedly invalid) certificates that used to be parseable 

### ISSUE THREAD

**Issue by pjsg** [bcgit/bc-java#1508]
We use bc-java to parse certificates but 1.75 fails to parse some invalid certificates (which are mostly valid). Ideally bc would return more information about the failure rather than just throw an IllegalArgumentException.

Suggest that a subclass of IllegalArgumentException be thrown with more information about why the certificate was invalid. 

I can make this change.....

**Comment by dghgit** [comment]
Yes, if you would like to submit a pull request, I'd be happy to look at it. Curious to know what the errors are, I'm guessing they may be about malformed integers?

**Comment by pjsg** [comment]
I stuck up a WIP PR: #1511 which is the functional code that implements the change. I haven't added tests for the changes yet (the actual tests that I have been using are in our private repo which makes use of BC). If this approach seems reasonable, then I'll go ahead and do that.

Some of the issues are malformed integers, but there are others as well -- invalid version numbers etc.

**Comment by winfriedgerlach** [comment]
@dghgit is this still being worked on?

**Comment by dghgit** [comment]
We've taken a slightly different angle on this, have added an X509CertificateReviewer to the org.bouncycastle.cert package which will give a verbose (hopefully not too verbose) report on a certificate it is given, in combination with ideas in PR #1151. Now up on https://www.bouncycastle.org/betas
